assets/js/plugin-check-admin.js

```diff
@@ -33,20 +39,62 @@
 		)
 		.then(
 			( data ) => {
-				if ( ! data ) {
-					throw new Error( 'Response contains no data' );
+				return handleDataErrors( data );
+			}
+		)
+		.then(
+			( data ) => {
+				console.log( data );
+
+				if ( ! data.data || ! data.data.plugin || ! data.data.checks ) {
+					throw new Error( 'Plugin and Checks are missing from the response.' );
 				}
 
-				if ( ! data.success ) {
-					// // If not successful and no message in the response.
-					if ( ! data.data || ! data.data[0].message ) {
-						throw new Error( 'Response contains no data' );
-					}
+				// Store the plugin and checks to run to be used later.
+				pluginChecker.plugin = data.data.plugin;
+				pluginChecker.checks = data.data.checks;
+
+				runChecks();
+			}
+		)
+		.catch(
+			( error ) => { console.error( error ) }
+		);
+	} );
```

Comment:
Can we move all this into a function to get the checks to run like you already did with the other code? Now that this file becomes more complex, we should start working with logical units.

I also think it would be best to return a `Promise` from the new function with the resulting data to decouple the two functions. This new function shouldn't set the data on the `pluginChecker` const, and it shouldn't call `runChecks()`. It should only return the data, and then the event listener can take care of the specific bits to connect the two.